test/collectors_test.lua

```diff
@@ -160,3 +154,114 @@ g.test_counter_cache = function()
     obs = utils.find_obs('cnt2', {}, observations)
     t.assertEquals(obs.value, 7, 'counter_3 is the only reference to cnt2')
 end
+
+g.test_global_labels = function()
+    --- Incorrect global label
+    ensure_throws("label key must be a string", function()
+        metrics.set_global_labels({ [2] = 'value' })
+    end)
+
+    --- Set correct global label
+    metrics.set_global_labels({ alias = 'my_instance' })
+
+    --- Ensure global label appends to counter observations
+    local counter = metrics.counter('counter', 'test counter')
+
+    -- Make some observation
+    counter:inc(3)
+
+    -- Collect metrics and check their labels
+    local observations = metrics.collect()
+    local obs_cnt = utils.find_obs('counter', { alias = "my_instance" }, observations)
+    t.assertEquals(obs_cnt.value, 3, "observation has global label")
```

Comment:
use snake case please

Reply:
I can provide renaming tool that will bring everything to snake case

Reply:
Will be done in a separate MR